controller/k8s/endpoints.go

```diff
@@ -395,43 +374,38 @@ func (sp *servicePort) updateService(newService *v1.Service) {
 	}
 	if newTargetPort != sp.targetPort {
 		newAddresses := addresses(sp.endpoints, newTargetPort)
-
-		log.Debugf("Updating %s:%d to %s", sp.service, sp.port, util.AddressesToString(newAddresses))
-
-		add, remove := util.DiffAddresses(sp.addresses, newAddresses)
-		for _, listener := range sp.listeners {
-			listener.Update(add, remove)
-		}
+		sp.updateAddresses(newAddresses)
 		sp.targetPort = newTargetPort
 		sp.addresses = newAddresses
 	}
 }
 
-func (sp *servicePort) deleteService() {
-	sp.mutex.Lock()
-	defer sp.mutex.Unlock()
-
-	newTargetPort := intstr.FromInt(int(sp.port))
-	if newTargetPort != sp.targetPort {
```

Comment:
As far as I can tell, the `deleteService` method was a no-op. The delete event returns the previous version of the service, so this if condition always returns false. It's also the case that when we delete a service, we receive both a DeleteEndpoints event and a DeleteService event, so I've removed the DeleteService handler, in favor of only having a DeleteEndpoints handler.